opm/material/fluidsystems/blackoilpvt/LiveOilPvt.hpp

```diff
@@ -59,20 +58,20 @@ class LiveOilPvt
     /*!
      * \brief Initialize the oil parameters via the data specified by the PVTO ECL keyword.
      */
-    void initFromDeck(const Deck& deck, const EclipseState& eclState)
+    void initFromDeck(const EclipseState& eclState)
     {
         const auto& pvtoTables = eclState.getTableManager().getPvtoTables();
-        const auto& densityKeyword = deck.getKeyword("DENSITY");
+        const auto& densityKeyword = eclState.getTableManager().getDensityTable();
 
         assert(pvtoTables.size() == densityKeyword.size());
 
         size_t numRegions = pvtoTables.size();
         setNumRegions(numRegions);
 
         for (unsigned regionIdx = 0; regionIdx < numRegions; ++ regionIdx) {
-            Scalar rhoRefO = densityKeyword.getRecord(regionIdx).getItem("OIL").getSIDouble(0);
-            Scalar rhoRefG = densityKeyword.getRecord(regionIdx).getItem("GAS").getSIDouble(0);
-            Scalar rhoRefW = densityKeyword.getRecord(regionIdx).getItem("WATER").getSIDouble(0);
+            Scalar rhoRefO = densityKeyword[regionIdx].oil;
```

Comment:
yes, performance is not relevant here.

here, you could want to get this number from some other object e.g. to let the user change it at runtime. If you now have multiple places where you store the densities, you may want them to all access the same number.

the point is: it's hard to say in advance what you need in a year and accessor methods provide some degree of flexibility.